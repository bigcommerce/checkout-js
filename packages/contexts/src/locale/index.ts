export { LocaleContext, type LocaleContextType } from './LocaleContext';
export { default as LocaleProvider } from './LocaleProvider';
export { useLocale } from './useLocale';
