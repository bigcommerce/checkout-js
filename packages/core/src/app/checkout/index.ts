export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
