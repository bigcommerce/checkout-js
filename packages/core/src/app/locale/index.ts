export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
