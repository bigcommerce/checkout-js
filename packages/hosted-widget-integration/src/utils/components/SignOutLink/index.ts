export { default as SignOutLink } from './SignOutLink';
