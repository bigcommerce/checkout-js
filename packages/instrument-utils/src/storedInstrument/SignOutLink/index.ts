export { default as SignOutLink, SignOutLinkProps } from './SignOutLink';
