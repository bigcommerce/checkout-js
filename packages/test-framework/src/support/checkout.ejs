<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
