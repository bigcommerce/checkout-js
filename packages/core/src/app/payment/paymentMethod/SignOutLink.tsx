import { PaymentMethod } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent } from 'react';

import { preventDefault } from '../../common/dom';
import { TranslatedString, withLanguage, WithLanguageProps } from '../../locale';

import getPaymentMethodName from './getPaymentMethodName';

export interface SignOutLinkProps {
    method: PaymentMethod;
    onSignOut(): void;
}

const SignOutLink: FunctionComponent<SignOutLinkProps & WithLanguageProps> = ({
    language,
    method,
    onSignOut,
}) => (
    <div className="signout-link">
        <TranslatedString id="remote.sign_out_before_action" />{' '}
        <a href="#" onClick={preventDefault(onSignOut)}>
            <TranslatedString
                data={{ providerName: getPaymentMethodName(language)(method) }}
                id="remote.sign_out_action"
            />
        </a>{' '}
        <TranslatedString id="remote.sign_out_after_action" />
    </div>
);

export default withLanguage(SignOutLink);
