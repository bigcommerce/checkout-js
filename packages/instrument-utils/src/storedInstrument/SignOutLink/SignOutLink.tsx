import { PaymentMethod } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent } from 'react';

import { TranslatedString, useLocale } from '@bigcommerce/checkout/locale';
import { getPaymentMethodName } from '@bigcommerce/checkout/payment-integration-api';
import { preventDefault } from '@bigcommerce/checkout/ui';

export interface SignOutLinkProps {
    method: PaymentMethod;
    onSignOut(): void;
}

const SignOutLink: FunctionComponent<SignOutLinkProps> = ({ method, onSignOut }) => {
    const { language } = useLocale();

    return (
        <div className="signout-link">
            <TranslatedString id="remote.sign_out_before_action" />{' '}
            {
                // eslint-disable-next-line jsx-a11y/anchor-is-valid, jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
                <a href="#" onClick={preventDefault(onSignOut)}>
                    <TranslatedString
                        data={{ providerName: getPaymentMethodName(language)(method) }}
                        id="remote.sign_out_action"
                    />
                </a>
            }{' '}
            <TranslatedString id="remote.sign_out_after_action" />
        </div>
    );
};

export default SignOutLink;
