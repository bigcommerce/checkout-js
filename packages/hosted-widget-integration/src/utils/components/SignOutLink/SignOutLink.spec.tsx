import { createLanguageService } from '@bigcommerce/checkout-sdk';
import { mount, render } from 'enzyme';
import { noop } from 'lodash';
import React from 'react';

import { LocaleContext, LocaleContextType } from '@bigcommerce/checkout/locale';
import { getPaymentMethod } from '@bigcommerce/checkout/test-utils';

import SignOutLink from './SignOutLink';

describe('SignOutLink', () => {
    let localeContext: LocaleContextType;

    beforeEach(() => {
        localeContext = {
            language: createLanguageService(),
        };
    });

    it('renders output that matches snapshot', () => {
        const component = render(
            <LocaleContext.Provider value={localeContext}>
                <SignOutLink method={getPaymentMethod()} onSignOut={noop} />
            </LocaleContext.Provider>,
        );

        expect(component).toMatchSnapshot();
    });

    it('triggers callback when it is clicked', () => {
        const handleSignOut = jest.fn();
        const component = mount(
            <LocaleContext.Provider value={localeContext}>
                <SignOutLink method={getPaymentMethod()} onSignOut={handleSignOut} />
            </LocaleContext.Provider>,
        );

        component.find('a').simulate('click');

        expect(handleSignOut).toHaveBeenCalled();
    });
});
