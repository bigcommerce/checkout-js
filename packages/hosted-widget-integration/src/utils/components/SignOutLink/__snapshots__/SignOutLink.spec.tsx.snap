// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOutLink renders output that matches snapshot 1`] = `
<div
  class="signout-link"
>
  optimized_checkout.remote.sign_out_before_action 
  <a
    href="#"
  >
    optimized_checkout.remote.sign_out_action
  </a>
   optimized_checkout.remote.sign_out_after_action
</div>
`;
